src/RefractDataStructure.cc

```diff
@@ -1289,76 +1336,188 @@ namespace
+        ConversionContext& context,
+        const mson::BaseTypeName defaultNestedType)
+    {
+        return MsonMemberToRefract<ValueTrait>(MakeNodeInfo(section, sourceMap->value),
+            context,
+            GetType(section.valueDefinition, context),
+            defaultNestedType);
+    }
 
-        return ref;
+    std::unique_ptr<IElement> MsonTypeSectionToElements( //
+        const mson::Element::MixinSection& section,
+        const snowcrash::SourceMap<mson::Element>* sourceMap,
+        ConversionContext&,
+        const mson::BaseTypeName)
+    {
+        auto ref = make_element<RefElement>(section.typeSpecification.name.symbol.literal);
+        ref->attributes().set(SerializeKey::Path, from_primitive(SerializeKey::Content));
+        return std::move(ref);
     }
 
-    std::unique_ptr<IElement> MsonElementToRefract(const NodeInfo<mson::Element>& mse,
+    std::unique_ptr<IElement> MsonTypeSectionToElements( //
+        const mson::Element::OneOfSection& section,
+        const snowcrash::SourceMap<mson::Element>* sourceMap,
         ConversionContext& context,
-        const mson::BaseTypeName defaultNestedType /* = mson::StringTypeName */)
+        const mson::BaseTypeName);
+
+    std::unique_ptr<IElement> MsonTypeSectionToElements( //
+        const mson::Element::OneOfSection& section,
+        const snowcrash::SourceMap<mson::Element>* sourceMap,
+        ConversionContext& context,
+        const mson::BaseTypeName)
     {
-        switch (mse.node->klass) {
-            case mson::Element::PropertyClass:
-                return MsonMemberToRefract<PropertyTrait>(
-                    MakeNodeInfo(mse.node->content.property, mse.sourceMap->property),
-                    context,
-                    GetType(mse.node->content.property.valueDefinition, context),
-                    defaultNestedType);
+        const NodeInfo<mson::OneOf>& oneOf = MakeNodeInfo(*section, sourceMap->oneOf());
 
-            case mson::Element::ValueClass:
-                return MsonMemberToRefract<ValueTrait>(MakeNodeInfo(mse.node->content.value, mse.sourceMap->value),
-                    context,
-                    GetType(mse.node->content.value.valueDefinition, context),
-                    defaultNestedType);
+        NodeInfoCollection<mson::OneOf> oneOfNodeInfo(oneOf);
+        auto select = oneOfNodeInfo.empty() ? make_empty<SelectElement>() : make_element<SelectElement>();
 
-            case mson::Element::MixinClass:
-                return MsonMixinToRefract(MakeNodeInfo(mse.node->content.mixin, mse.sourceMap->mixin));
+        for (const auto& oneOfInfo : oneOfNodeInfo) {
+            oneOfInfo.node->visit(MsonOneOfEntryToElementsLambda{ sourceMap, context, *select });
+        }
 
-            case mson::Element::OneOfClass:
-                return MsonOneofToRefract(MakeNodeInfo(mse.node->content.oneOf(), mse.sourceMap->oneOf()), context);
+        return std::move(select);
```

Comment:
some special reason for `return std::move()` generally it is better to avoid because of copy elision

https://en.wikipedia.org/wiki/Copy_elision#Return_value_optimization

Reply:
Oh it seem, it is rest of some older changes.
Please remove it separate commit in whole file

Reply:
I believe gcc 4.8.5 complained (incomplete implementation of return value elision). Cool that you have noticed this though.

Reply:
Maybe we can wrap it in a macro so that we can prevent move in modern compilers (and also prevent compiler warnings in newer compilers):

```c
#if GCC_VERSION > 4.9
#define MOVE(x) x
#else
#define MOVE(x) std::move(x)
#endif

return MOVE(x)
```

Reply:
In hindsight - I was wrong. The compiler was right. A better way to avoid copies is to forbid them.

That's not in the scope of this PR though, so I'm removing the `move`s relying on optional RVO in compilers; it's around since C++98 and from C++17 onwards, RVO is even required in most of these cases.